### DISCUSSION TITLE
How to get the droplet trajectory？

### DISCUSSION THREAD

**Discussion by RachelMeng0096** [AMReX-Combustion/PeleC#781] [Q&A] [answered]
![image](https://github.com/AMReX-Combustion/PeleC/assets/153271449/024ab58a-3af3-40c4-b01d-820a2ce8a6e9)
As the figure aboved,I'm interested in tracking a droplet to obtain its trajectory line. However, I find myself somewhat at a loss on how to proceed with this task. Would you be able to provide me with some advice or suggestions? Your guidance would be greatly appreciated！

**Comment by baperry2** [comment, marked as answer]
I'm not sure on this as I haven't actually done it, but here's something you can try.

To track spray droplets you should be able to run with `particles.write_ascii_files = 1`, which will output a text file with particle locations and data at the same intervals as you are dumping plt/chk files. These files are unlabelled but you can deduce what the columns mean based on [how they are written](https://github.com/AMReX-Codes/amrex/blob/64d2360b209c1e625bfc9a6feeb7501a618f7ed5/Src/Particle/AMReX_ParticleIO.H#L1239). The first AMREX_SPACEDIM columns are the location, the next several columns contain the particle data as defined [here](https://github.com/AMReX-Combustion/PelePhysics/blob/80f2cfa0edd2e928a93bb9dff45e68d32d8699f7/Source/Spray/SprayFuelData.H#L10), then the final two columns are the particle id and processor id.

According to the [AMReX documentation](https://amrex-codes.github.io/amrex/docs_html/Particle.html#id5), the combination of particle id and processor (the processor on which the particle was generated, not where it is) should be unique and allow you to track particles over time.

**Reply by RachelMeng0096** [reply]
Thank you very much for your response. I also managed to track the droplets I needed in Paraview using a similar method. I am very grateful for your help.
